plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Repository.java

```diff
@@ -244,13 +162,8 @@ The default behaviour is to detect which access style to use based on the config
         AmazonS3 client = s3Service.client(metadata.settings());
         blobStore = new S3BlobStore(settings, client, bucket, serverSideEncryption, bufferSize, cannedACL, storageClass);
 
-        String basePath = getValue(metadata.settings(), settings, Repository.BASE_PATH_SETTING, Repositories.BASE_PATH_SETTING);
+        String basePath = BASE_PATH_SETTING.get(metadata.settings());
         if (Strings.hasLength(basePath)) {
-            if (basePath.startsWith("/")) {
-                basePath = basePath.substring(1);
-                deprecationLogger.deprecated("S3 repository base_path trimming the leading `/`, and " +
-                                                 "leading `/` will not be supported for the S3 repository in future releases");
-            }
             this.basePath = new BlobPath().add(basePath);
         } else {
             this.basePath = BlobPath.cleanPath();
```

Comment:
Not related to this change but method `public static <T> T getValue(...)` at the end of this class can become package private.